### PR TITLE
Add GroundingSAM (GroundingDino + SAM)

### PR DESCRIPTION
**Pull request by pavel-esir**
ticket: CVS-131290

### CODE REVIEW

**Comment by pavel-esir** [comment]
<p>very obsolete name, i'll rename 🤦</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1720/discussion/#comment-1964457275'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pavel-esir** [comment]
<p>Torch 2.2 tvision 0.17 works as well, i will remove manual install of torch and will rely on timm with extra index url.</p><p> </p><p>About pycocotools&nbsp;it also seemed strange to me, but without it i end up having import error, because it's needed in visualizer</p><p><a href="https://github.com/wenyi5608/GroundingDINO/blob/main/groundingdino/util/visualizer.py#L19" target="_blank">https://github.com/wenyi5608/GroundingDINO/blob/main/groundingdino/util/visualizer.py#L19</a></p><p>which is imported in file where model torch nn.Module is defined</p><p><a href="https://github.com/wenyi5608/GroundingDINO/blob/main/groundingdino/models/GroundingDINO/groundingdino.py#L37" target="_blank">https://github.com/wenyi5608/GroundingDINO/blob/main/groundingdino/models/GroundingDINO/groundingdino.py#L37</a></p><p> </p><p>Supervision is needed for simpler annotation, but it can be done manually. I will do that and will remove dependency</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1720/discussion/#comment-1964457183'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pavel-esir** [comment]
<p>i have drawn masks manually in PIL but it turned out supervision is still imported inside for post processing <code>groundingdino.util.inference.Model</code></p><p>i will rewrite this part and will remove supervision dependency</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1720/discussion/#comment-1964457183'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by eaidova** [comment]
> i will rewrite this part and will remove supervision dependency

@pavel-esir  it is not a bad to have it in  deps if we know why this package used, so if it is difficult and required additional time for rewriting, I'm ok to stay it

**Comment by eaidova** [comment]
@pavel-esir  looks like you accidentally commited changes in 261 notebook, could you please remove them?

**Comment by pavel-esir** [comment]
<p>replaced with gr.Interface and and specified examples arg, this gives UI similar to as gr.Examples</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1720/discussion/#comment-1964457371'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pavel-esir** [comment]
<p>moved them outside</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1720/discussion/#comment-1967934215'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pavel-esir** [comment]
<p>done</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1720/discussion/#comment-1967934223'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pavel-esir** [comment]
<p>thanks a lot. Will apply this</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1720/discussion/#comment-1968950841'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pavel-esir** [comment]
<p>done</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1720/discussion/#comment-1968950841'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pavel-esir** [comment]
<p>Returned back supervision. With it code looks simpler.</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1720/discussion/#comment-1964457183'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pavel-esir** [comment]
<p>done, significantly enhanced providing example input</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1720/discussion/#comment-1964457213'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pavel-esir** [comment]
<p>i will provide description for this function as well as description with running combined GroundingDINO + SAM/EfficientSAM</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1720/discussion/#comment-1964457247'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pavel-esir** [comment]
<p>removed</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1720/discussion/#comment-1968950858'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pavel-esir** [comment]
<p>removed</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1720/discussion/#comment-1967934210'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pavel-esir** [comment]
<p>agreed and added clarifications</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1720/discussion/#comment-1964457338'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pavel-esir** [comment]
<p>slightly refactored this function, added comments with explanations, and type annotatetions</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1720/discussion/#comment-1964457247'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pavel-esir** [comment]
<p>added clarification for this part</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1720/discussion/#comment-1964457305'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pavel-esir** [comment]
Applied comments. @eaidova @apaniukov @aleksandr-mokrov please take a look

**Comment by pavel-esir** [comment]
compilation on mac fails. i will disable testing on mac

**Comment by eaidova** [comment]
@pavel-esir could you please check correctness of notebook meta for docs?

image

I'm not sure that visual question answering is correct task for this notebook, also not sure that it is possible to run it using binder (binder has only 2GB RAM, are you sure that it is enough for running your notebook? do you try to run it?)

Please also add image for preview

**Comment by pavel-esir** [comment]
<p>done</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1720/discussion/#comment-1985047038'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pavel-esir** [comment]
<p>done</p>

---

 <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1720/discussion/#comment-1985050445'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by pavel-esir** [comment]
@added link to image, removed visual question answering, applied.

>not sure that it is possible to run it using binder (binder has only 2GB RAM, are you sure that it is enough for running your notebook? do you try to run it?)

checked, indeed it consumes too much ram > 5Gb, removed binder link

**Review by eaidova — APPROVED** [review]

File: notebooks/288-grounded-segment-anything/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Object detection and masking from prompts with GroundedSAM (GroundingDINO + SAM) and OpenVINO
+
+[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/openvinotoolkit/openvino_notebooks/HEAD?filepath=notebooks%2F288-grounded-segment-anything%2F288-grounded-segment-anything.ipynb)
```
Comment (pavel-esir) [inline notebooks/288-grounded-segment-anything/README.md:3]:
```suggestion
```
